app/models/discovery_rule.rb

```diff
@@ -36,6 +36,13 @@ class DiscoveryRule < ApplicationRecord
     end
   }
 
+  def unique_priority_per_organization
+    discovery_rule_ids  =  TaxableTaxonomy.where(taxable_type: 'DiscoveryRule', taxonomy_id: organization_ids).pluck(:taxable_id)
```

Comment:
it's better but still ignores nested orgs since it's using organization_ids only

Reply:
There are also two extra spaces, just a nitpick.

Reply:
Rahuj just use business layer methods to return all organizations associated and then get all their ids.

Reply:
Scratch that comment, query is the proper way of finding IDs.

Reply:
@ares @lzap i tried to write a more generic method, using the `Taxonomy.enabled_taxonomies` but when iterating with the location_ids i am facing some issues, which i am not being able to understand. There is the code that i tried:
```
  def unique_priority
    Taxonomy.enabled_taxonomies.each do |enabled_taxonomy|
      discovery_rule_ids  =  TaxableTaxonomy.where(taxable_type: 'DiscoveryRule', taxonomy_id: send("#{enabled_taxonomy.singularize}_ids")).pluck(:taxable_id)
      if DiscoveryRule.find(discovery_rule_ids).pluck(:priority).include?(priority)
        errors.add(:priority, _("is already taken"))
      end
    end
  end
```